airflow/models/taskinstance.py

```diff
@@ -2529,6 +2536,28 @@ def filter_for_tis(tis: Iterable[Union["TaskInstance", TaskInstanceKey]]) -> Opt
             (ti.key.primary for ti in tis),
         )
 
+    @classmethod
+    def filter_for_task_id_map_index_lists(cls, vals):
+        """
+        Build an SQLAlchemy filter for a list where each element can contain
+        whether a task_id, or a tuple of (task_id,map_index)
+
+        :meta private:
+        """
+        # Compute a filter for TI.task_id and TI.map_index based on input values
+        # For each item, it will either be a task_id, or (task_id, map_index)
+        task_id_only = list(filter(lambda v: isinstance(v, str), vals))
+        with_map_index = list(filter(lambda v: not isinstance(v, str), vals))
+        filters = []
+
+        if task_id_only:
+            filters.append(cls.task_id.in_(task_id_only))
+        if with_map_index:
+            filters.append(
+                tuple_in_condition((cls.task_id, cls.map_index), with_map_index),
+            )
+        return or_(*filters) if len(filters) > 1 else filters[0]
```

Comment:
```suggestion
        if not filters:
            return true()
        elif len(filters) == 1:
            return filters[0]
        return or_(*filters)
```

Calling `or_()` without any arguments is deprecated https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.or_

Reply:
In my head this function should never result in 0 filters. I should assert that